### PR TITLE
 XIVY-10351 Create TaskTab

### PR DESCRIPTION
**Pull request by ivy-lli**
- Add checkbox widget (persist option, code tab: disable persmission)
- Add select widget
- Add ability to resize CodeEditor widget (code tab)
- Refactor table parts (and add SelectCell)

### CODE REVIEW

File: client/packages/core/src/data/task-data.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    persist: boolean;
+    task: {
+      name: string;
+      description: string;
+      category: string;
+      priority: string;
+      expiry: {
+        timeout: string;
+        error: string;
+        responsible: {
```
Comment (ivy-rew) [inline client/packages/core/src/data/task-data.ts:32]: I think we should align here: as @ivy-lmu is soon implementing this in the backed: 
I thought that another structure may be easier:

```
responsible: {
  mode = "role/user/..." <-- enum
  assignee = "Everybody/System/... " or "in.myUser" ...
}
```
What do you think about it?

Comment (ivy-lli) [reply client/packages/core/src/data/task-data.ts:32]: Yes this is the current state from the backend.
And yes something you proposed should be the future :+1:

File: client/packages/editor/src/components/tabs/CodeTab.tsx [hunk 2]
```diff
@@ -2,18 +2,28 @@ import { useData } from '../../context';
 import { TabProps, TabState } from '../props';
 import LabelInput from '../widgets/label/LabelInput';
 import { CodeEditor } from '../widgets';
+import Checkbox from '../widgets/checkbox/Checkbox';
+
+const SECURITY_SYSTEM = 'system';
```
Comment (ivy-rew) [inline client/packages/editor/src/components/tabs/CodeTab.tsx:7]: I have a task for that: hopefully the client soon can forget this special value :relaxed: 

i think we will introduce:
`checkPermissions = true/false`
or if someone has a preference:
`asSystem = true/false`

File: client/packages/editor/src/components/widgets/select/ResponsibleSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  return (
+    <div className='responsible-select'>
+      <Select label='Responsible' items={responsibleItems} value={selectedResponsible} onChange={item => setResponsible(item.value)}>
+        <>
+          {selectedResponsible?.value === 'ROLE' && <Select label='Role' items={roleItems} value={selectedRole} onChange={() => {}} />}
+          {selectedResponsible?.value === 'ROLE_FROM_ATTR' && <input className='input' />}
+          {selectedResponsible?.value === 'USER_FROM_ATTR' && <input className='input' />}
+        </>
+      </Select>
```
Comment (ivy-rew) [inline client/packages/editor/src/components/widgets/select/ResponsibleSelect.tsx:32]: things look so easy if solved with REACT+TS ... this component was much more complex back in the SWT days :+1:

File: client/packages/editor/src/components/widgets/select/Select.css [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  gap: 0.3rem;
+  width: 100%;
+}
+.select-button:focus {
+  box-shadow: var(--focus-shadow);
+}
+.select-menu {
+  overflow: auto;
+  width: 100%;
+  max-height: 20rem;
```
Comment (ivy-rew) [inline client/packages/editor/src/components/widgets/select/Select.css:26]: I'm totally outdated in CSS ... never heard of 'rem' values

File: client/packages/editor/src/components/widgets/table/DocumentTable.tsx [hunk 5]
```diff
@@ -89,31 +88,19 @@ const DocumentTable = (props: { data: Document[]; onChange: (change: Document[])
...
-                </button>
-                <button aria-label='Browse'>
-                  <IvyIcon icon={IvyIcons.Search} />
-                </button>
-                <button aria-label='Open URL'>
-                  <IvyIcon icon={IvyIcons.GoToSource} />
-                </button>
-              </span>
-            </TableCell>
+            <ActionCell
```
Comment (ivy-rew) [inline client/packages/editor/src/components/widgets/table/DocumentTable.tsx:91]: cool generalization in here ...

File: client/packages/editor/src/components/widgets/table/header/TableHeader.css [hunk 6]
```diff
@@ -0,0 +1,11 @@
+.table .table-column-header {
+  padding: 10px;
+  text-align: start;
+  font-size: 10px;
```
Comment (ivy-rew) [inline client/packages/editor/src/components/widgets/table/header/TableHeader.css:4]: maybe we should avoid absolute values in a concrete widget ... making the inscription editor easier to access on any device ... even in SWT the user could choose the font-size for the editors. If we go back to static values, we will loose users with bad eye-sight :)

**Review by ivy-rew — APPROVED** [review]
looks good :100:

File: client/packages/editor/src/components/tabs/CodeTab.tsx (hunk 2 above)
Comment (ivy-lli) [reply client/packages/editor/src/components/tabs/CodeTab.tsx:7]: both are good for me. 
Maybe the one which is more the truth? I mean does this flag mean that there are no permission checks (but the same user or is the script step run with the system user?

File: client/packages/editor/src/components/widgets/select/Select.css (hunk 4 above)
Comment (ivy-lli) [reply client/packages/editor/src/components/widgets/select/Select.css:26]: root element font size. mostly in browsers 16px but can be changed by users. So if a user uses a bigger font-size also this rem value increases.

File: client/packages/editor/src/components/widgets/table/header/TableHeader.css (hunk 6 above)
Comment (ivy-lli) [reply client/packages/editor/src/components/widgets/table/header/TableHeader.css:4]: yes I need to change this px values to rem or to another responsive unit :)

File: client/packages/core/src/data/task-data.ts (hunk 1 above)
Comment (ivy-lli) [reply client/packages/core/src/data/task-data.ts:32]: the first one. It's ok if you use the second one to save, but it's really difficult to check which value is available and what to render for it...

Comment (ivy-rew) [reply client/packages/core/src/data/task-data.ts:32]: good, yes I think the first one would be the best. Naturally users will think of the data being structured as it is presented on the UI ... keeping the 'mode' and the 'assignee' separate, is also on the plain JSON easier to use. As it is less easy to setup a corrupt configuration :)

File: client/packages/editor/src/components/widgets/select/Select.css (hunk 4 above)
Comment (ivy-rew) [reply client/packages/editor/src/components/widgets/select/Select.css:26]: thanks for the update :love_letter:  @ivy-lli

**Comment by ivy-lmu** [comment]
Cool 👍

**Pull request merged**